### PR TITLE
[13.0][MIG] product_assortment

### PR DESCRIPTION
**Pull request by Laurent-Corron**

### CODE REVIEW

File: product_assortment/views/product_assortment.xml [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+                        </button>
+                    </div>
+                    <group col="4">
+                        <field name="name" string="Assortment Name"/>
+                        <field name="model_id" invisible="1"/>
+                        <field name="active"/>
+                        <field name="is_assortment" invisible="1"/>
+                    </group>
+                    <group>
+                        <field name="domain" widget="char_domain" options="{'model_field': 'model_id'}"/>
```
Comment (lmignon) [inline product_assortment/views/product_assortment.xml]: @Laurent-Corron The widget to use is not properly set...

```suggestion
                        <field name="domain" widget="domain" options="{'model': 'product.product'}"/>
```

**Review by lmignon — CHANGES_REQUESTED** [review]

File: product_assortment/views/product_assortment.xml (hunk 1 above)
Comment (Laurent-Corron) [reply product_assortment/views/product_assortment.xml]: fixed!

**Review by lmignon — APPROVED** [review]
Code review + functional tests

**Review by rousseldenis — APPROVED** [review]
Code review

File: product_assortment/models/ir_filters.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
+# Copyright 2018-2019 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import _, api, fields, models
+from odoo.osv import expression
+
+
+class ProductAssortment(models.Model):
```
Comment (legalsylvain) [inline product_assortment/models/ir_filters.py]: class IrFilters(models.Model):

File: product_assortment/models/ir_filters.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    whitelist_product_ids = fields.Many2many(
+        comodel_name="product.product", relation="assortment_product_whitelisted"
+    )
+
+    record_count = fields.Integer(compute="_compute_record_count")
+
+    is_assortment = fields.Boolean(default=lambda x: x._get_default_is_assortment())
+
+    def _get_eval_domain(self):
+        res = super(ProductAssortment, self)._get_eval_domain()
```
Comment (legalsylvain) [inline product_assortment/models/ir_filters.py]: res = super()._get_eval_domain()

File: product_assortment/models/ir_filters.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    def _compute_record_count(self):
+        for record in self:
+            domain = record._get_eval_domain()
+            record.record_count = self.env[record.model_id].search_count(domain)
+
+    @api.model
+    def _get_action_domain(self, action_id=None):
+        # tricky way to act on get_filter method to prevent returning
+        # assortment in search view filters
+        domain = super(ProductAssortment, self)._get_action_domain(action_id=action_id)
```
Comment (legalsylvain) [inline product_assortment/models/ir_filters.py]: domain = super()._get_action_domain(action_id=action_id)

File: product_assortment/models/ir_filters.py [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+class ProductAssortment(models.Model):
+    _inherit = "ir.filters"
+
+    @api.model
+    def _get_default_model(self):
+        if self.env.context.get("product_assortment", False):
+            model = self.env.ref("product.model_product_product")
+            return model.model
+        return ""
```
Comment (legalsylvain) [inline product_assortment/models/ir_filters.py]: return False

**Review by legalsylvain — APPROVED** [review]
Hi. 

Thanks for porting this module. Some minor remarks inline. The rest LGTM for me. Code review / no test.

BTW, I'm just discovering this module. i think it is great, but should be more generic. This feature is interesting for product, but could be also implemented for partners, or any other model.

A more abstract module, in server-ux could be interesting.

not a blocking point, just sharing my pov.

kind regards.

**Comment by lmignon** [comment]
@OCA/product-maintainers This one is ready to merge. Thank you!

**Comment by lmignon** [comment]
@OCA/product-maintainers This one is ready to merge. Thank you!

**Comment by lmignon** [comment]
@legalsylvain This one is ready to merge...

**Comment by rousseldenis** [comment]
@legalsylvain

**Pull request merged**